src/Feldspar/Core/Middleend/FromTyped.hs

```diff
@@ -144,7 +144,7 @@ trOp R.Equal           = Equal
 trOp R.NotEqual        = NotEqual
 trOp R.Undefined       = Undefined
 trOp (R.Assert s)      = Assert s
--- trOp R.ForeignImport   = ForeignImport
+trOp (R.ForeignImport s _) = ForeignImport s
```

Comment:
Is `R.Call` a few lines down involved in this somehow?

Reply:
I do not think so. The way I read the old Feldspar is that 'ForeignImport s f' becomes a variadic symbol that is combined with its arguments using the constructor :@. This also works with the evaluator (in Semantics.hs). So Call is not involved.